hw/xbox/xid.c

```diff
@@ -215,19 +222,29 @@ static void usb_xid_handle_control(USBDevice *dev, USBPacket *p,
 {
     USBXIDState *s = DO_UPCAST(USBXIDState, dev, dev);
 
-    DPRINTF("xid handle_control %x %x\n", request, value);
-
+    static unsigned int pid = 0;
+    static size_t bytes = 0;
+    DPRINTF("xid handle_control 0x%x 0x%x, packet %zu, byte %d\n", request, value, ++pid, bytes += length);
+#if 0
+    if ((request == USB_REQ_SET_CONFIGURATION) && (value == 1)) {
+        /* The OpenXDK code will try to set configuration 1 on every read */
+        DPRINTF("xid handled by configuration hack\n");
+        return;
+    }
+#endif
```

Comment:
It's true for many compiled OpenXDK applications.
This hack should still work.

Keep in mind that I wrote this XID code back in mid 2014 or something and can't remember the specifics of this "hack".
At least it's a rather clean "hack" and we can still remove it at a later point